app/jobs/verify_fixity_job.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+class VerifyFixityJob < ApplicationJob
+  queue_as Atc::Queues::VERIFY_FIXITY
+
+  def perform(stored_object_id)
+    stored_object = StoredObject.find(stored_object_id)
+
+    # For now, handle only AWS fixity check
+    return unless stored_object.storage_provider.aws?
+
+    existing_fixity_verification_record = FixityVerification.find_by(stored_object: stored_object)
+    return if existing_fixity_verification_record&.pending?
+
+    if existing_fixity_verification_record
+      process_existing_fixity_verification_record existing_fixity_verification_record
+    end
+    fixity_verification_record = create_pending_fixity_verification stored_object
+    provider_fixity_check = instantiate_provider_fixity_check fixity_verification_record
+    verify_fixity(fixity_verification_record, provider_fixity_check)
+  end
+
+  def process_existing_fixity_verification_record(existing_fixity_verification_record)
+    return if existing_fixity_verification_record.pending?
```

Comment:
Maybe this line isn't necessary, since we only call `process_existing_fixity_verification_record` after checking if the record is pending?

Reply:
It is redundant, but I'd rather leave it for now. Once the method is fleshed out, it can be removed.